provider/adapters/src/fallback/mod.rs

```diff
@@ -211,13 +246,23 @@ impl LocaleFallbacker {
     where
         P: DataProvider<LocaleFallbackLikelySubtagsV1Marker>
             + DataProvider<LocaleFallbackParentsV1Marker>
+            + DataProvider<CollationFallbackSupplementV1Marker>
             + ?Sized,
     {
         let likely_subtags = provider.load(Default::default())?.take_payload()?;
         let parents = provider.load(Default::default())?.take_payload()?;
+        let collation_supplement = match DataProvider::<CollationFallbackSupplementV1Marker>::load(
+            provider,
+            Default::default(),
+        ) {
+            Ok(response) => Some(response.take_payload()?),
+            // It is expected that not all keys are present
+            Err(_) => None,
+        };
```

Comment:
If we're discarding the error anyway, you can use `Result::ok` and `and_then`:

```suggestion
        let collation_supplement = DataProvider::<CollationFallbackSupplementV1Marker>::load(
            provider,
            Default::default(),
        ).and_then(DataResponse::take_payload).ok();
```

However, we should really have a special error type that we ignore, so that we don't accidentally ignore fundamental errors.

Reply:
Good catch; we should be checking for `MissingDataKey` only